tox.ini

```diff
@@ -1,144 +1,91 @@
 [tox]
-envlist = py3, xenial-dev, flake8, pylint, black, isort
+envlist = py3, lowest-supported, flake8, pylint, black, isort
 recreate = True
 
 [testenv]
-commands = {envpython} -m pytest {posargs:tests/unittests}
+basepython = python3
 setenv =
     LC_ALL = en_US.utf-8
 passenv=
     PYTEST_ADDOPTS
 
-[testenv:flake8]
-basepython = python3
+[flake_env]
+envdir = {toxworkdir}/.flake_env
 deps =
     flake8==3.9.2
-commands = {envpython} -m flake8 {posargs:cloudinit/ tests/ tools/ setup.py}
-
-# https://github.com/gabrielfalcao/HTTPretty/issues/223
-setenv =
-    LC_ALL = en_US.utf-8
-
-[testenv:pylint]
-basepython = python3
-deps =
-    # requirements
     pylint==2.11.1
-    # test-requirements because unit tests are now present in cloudinit tree
+    black==21.12b0
+    isort==5.10.1
     -r{toxinidir}/test-requirements.txt
     -r{toxinidir}/integration-requirements.txt
+
+[testenv:flake8]
+envdir = {[flake_env]envdir}
+deps = {[flake_env]deps}
+commands = {envpython} -m flake8 {posargs:cloudinit/ tests/ tools/ setup.py}
+
+[testenv:pylint]
+envdir = {[flake_env]envdir}
+deps = {[flake_env]deps}
 commands = {envpython} -m pylint {posargs:cloudinit tests tools}
 
 [testenv:black]
-basepython = python3
-deps =
-    # requirements
-    black==21.12b0
+envdir = {[flake_env]envdir}
+deps = {[flake_env]deps}
 commands = {envpython} -m black . --check
 
 [testenv:isort]
-basepython = python3
-deps =
-    isort==5.10.1
+envdir = {[flake_env]envdir}
+deps = {[flake_env]deps}
 commands = {envpython} -m isort . --check-only
 
 [testenv:format]
-basepython = python3
-deps =
-    black==21.12b0
-    isort==5.10.1
+envdir = {[flake_env]envdir}
+deps = {[flake_env]deps}
 commands =
     {envpython} -m isort .
     {envpython} -m black .
 
 [testenv:py3]
-basepython = python3
 deps =
     -r{toxinidir}/test-requirements.txt
 commands = {envpython} -m pytest \
             --durations 10 \
             {posargs:--cov=cloudinit --cov-branch \
             tests/unittests}
 
-[testenv:py27]
-basepython = python2.7
-deps = -r{toxinidir}/test-requirements.txt
+[testenv:lowest-supported]
```

Comment:
I don't think we can do away with a shared lowest-supported-deps definition just yet. 
Since `lowest-supported` is included in our default envlist it'll be run automatically in developer environments via `tox -p auto`. Without continuing to "pin" `httpretty=0.9.5` in a lowest-supported-dev target we'll fail to pass locally on focal, hirsute and I think impish too.

It'd like to ensure developers can run this build env to avoid getting a suprirse if CI runs it and fails. What do you think?